wsd/ClientRequestDispatcher.cpp

```diff
@@ -990,6 +1003,155 @@ void ClientRequestDispatcher::handleWopiDiscoveryRequest(
+        { CheckStatus::HostNotFound, "HOST_NOT_FOUND" },
+        { CheckStatus::HostUnReachable, "HOST_UNREACHABLE" },
+        { CheckStatus::UnspecifiedError, "UNSPECIFIED_ERROR" },
+        { CheckStatus::ConnectionAborted, "CONNECTION_ABORTED" },
+        { CheckStatus::ConnectionRefused, "CONNECTION_REFUSED" },
+        { CheckStatus::InvalidCertificate, "INVALID_CERTIFICATE" },
+        { CheckStatus::NotHttps, "NOT_HTTPS" },
+        { CheckStatus::NoScheme, "NO_SCHEME" },
+        { CheckStatus::Timeout, "TIMEOUT" }
+    };
+
+    CheckStatus result = CheckStatus::Ok;
+
+    if (uri.getScheme().empty())
+    {
+        result = CheckStatus::NoScheme;
+    }
+    else if (uri.getScheme() != "https")
+    {
+        result = CheckStatus::NotHttps;
+    }
+    else
+    {
+        // request the url
+        try
+        {
+            const auto hostAddress(Poco::Net::DNS::resolve(uri.getHost()));
+
+            Poco::Net::HTTPSClientSession httpSession(uri.getHost(), 443);
+            httpSession.setConnectTimeout(Poco::Timespan(0, 300));
+            Poco::Net::HTTPRequest httpRequest(Poco::Net::HTTPRequest::HTTP_GET,
+                                               uri.getPathAndQuery());
+            httpSession.sendRequest(httpRequest);
+            Poco::Net::HTTPResponse response;
+            std::istream* responseStream = &httpSession.receiveResponse(response);
+
+            std::cout << responseStream->rdbuf();
+            if (response.getStatus() != 200)
+            {
+                result = CheckStatus::NotHttpSucess;
+            }
+        }
+        catch (Poco::Net::HostNotFoundException& hostNotfound)
+        {
+            result = CheckStatus::HostNotFound;
+        }
+        catch (Poco::Net::NoAddressFoundException& noAddressFound)
+        {
+            result = CheckStatus::HostUnReachable;
+        }
+        catch (Poco::Net::ConnectionAbortedException& connectionAborted)
+        {
+            result = CheckStatus::ConnectionAborted;
+        }
+        catch (Poco::Net::ConnectionRefusedException& exception)
+        {
+            result = CheckStatus::ConnectionRefused;
+        }
+        catch (Poco::Net::InvalidCertificateException& invalidCertification)
+        {
+            result = CheckStatus::InvalidCertificate;
+        }
+        catch (Poco::Net::CertificateValidationException& certificateValidation)
+        {
+            result = CheckStatus::CertificateValidation;
+        }
+        catch (Poco::TimeoutException& timeout)
+        {
+            result = CheckStatus::Timeout;
+        }
+        catch (Poco::Exception& exception)
+        {
+            LOG_ERR_S("Wopi Access Check request error, query to callback ["
+                      << uri.toString() << "] failed:" << exception.what() << exception.className()
+                      << exception.name() << exception.message());
+
+            result = CheckStatus::UnspecifiedError;
+        }
+    }
```

Comment:
As far as I see this only checks the connectivity, ideally it would also verify the wopi aliasgroups if the host is in the allow list and provide a proper error code if that fails